### PR TITLE
i18n MessageController now supports config parameter 'markUnused' that turns adding @@s either on or off

### PR DESCRIPTION
**Pull request by marius7383**
This is a rework of pull request #8632 since it was not setup properly and addresses issue #8625 as well. 

@klimov-paul  About the newly added content in the upgrade-log I am not fully sure, because I had to add the `Upgrade from Yii 2.0.4` section and the `MessageController` doesn't seem to be a class that is supposed to be used in application development. Hope it's still correct.

### CODE REVIEW

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Yii Framework 2 Change Log
 - Enh #8486: Added support to automatically set the `maxlength` attribute for `Html::activeTextArea()` and `Html::activePassword()` (klimov-paul)
 - Enh #8566: Added support for 'only' and 'except' options for `yii\web\AssetManager::publish()` (klimov-paul)
 - Enh #8574: Added `yii\console\controllers\MessageController` support .pot file creation (pgaultier)
+- Enh #8625: MessageController adds "@@" to messages, that aren't directly used via Yii::t() (marius7383)
```
Comment (klimov-paul) [inline framework/CHANGELOG.md]: Should be:
```
Added 'markUnused' option to `yii\console\controllers\MessageController`
```

File: framework/UPGRADE.md [hunk 2]
```diff
@@ -8,6 +8,16 @@ if you want to upgrade from version A to version C and there is
 version B between A and C, you need to following the instructions
 for both A and B.
 
+
+Upgrade from Yii 2.0.4
+----------------------
+  
+* The signature of several `saveMessagesTo...` methods in `yii\console\controllers\MessageController` is changed. They now have an additional parameter named `markUnused`:
```
Comment (klimov-paul) [inline framework/UPGRADE.md]: Should be:
```
The signature of the following methods in `yii\console\controllers\MessageController` has changed:
```
Better will be following `yii\base\View::beforeRender()` and `yii\base\View::afterRender()` note at:
https://github.com/yiisoft/yii2/blob/master/framework/UPGRADE.md#upgrade-from-yii-20-rc

**Comment by klimov-paul** [comment]
Overall PR looks fine. Still there is a minor BC break.
@yiisoft/core-developers, what is your opinion: should we include this to 2.0.5 ?

File: framework/console/controllers/MessageController.php [hunk 3]
```diff
@@ -227,10 +230,14 @@ protected function saveMessagesToDb($messages, $db, $sourceMessageTable, $messag
...
+                if ($markUnused) {
+                    $expression = "CONCAT('@@',message,'@@')";
+                }
                 $db->createCommand()
                    ->update(
                        $sourceMessageTable,
-                       ['message' => new \yii\db\Expression("CONCAT('@@',message,'@@')")],
+                       ['message' => new \yii\db\Expression($expression)],
                        ['in', 'id', $obsolete]
                    )->execute();
```
Comment (cebe) [inline framework/console/controllers/MessageController.php]: the whole update statement is not needed if we should not mark unused.

**Comment by cebe** [comment]
If the `$markUnused` parameter was added at the end, the BC break would be okay for me as the param would be simply ignored. This way it will result in weird behavior.

**Pull request closed without merging**